files/unarchive.py

```diff
@@ -300,6 +300,16 @@ def main():
     if not os.access(src, os.R_OK):
         module.fail_json(msg="Source '%s' not readable" % src)
 
+    # skip working with 0 size archives
+    try:
+        if os.path.getsize(src) == 0:
+            res_args = {
+                'changed': False
```

Comment:
I would add msg="0 byte archive found, nothing to do", so users are not puzzled by the lack of change.